lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1228,6 +1228,65 @@ static bool doesAccessorNeedDynamicAttribute(AccessorDecl *accessor) {
   llvm_unreachable("covered switch");
 }
 
+llvm::Expected<CtorInitializerKind>
+InitKindRequest::evaluate(Evaluator &evaluator, ConstructorDecl *decl) const {
+  auto &diags = decl->getASTContext().Diags;
+
+  // Convenience inits are only allowed on classes and in extensions thereof.
+  if (decl->getAttrs().hasAttribute<ConvenienceAttr>()) {
+    if (auto extType = decl->getDeclContext()->getDeclaredInterfaceType()) {
```

Comment:
I suggest using

```
if (auto nominal = decl->getDeclContext()->getSelfNominalTypeDecl()) {
```

here instead of `getDeclaredInterfaceType()`, because it's cheaper to evaluate (name lookup vs. type checking). You can then `dyn_cast<ClassDecl>(nominal)` to get `extClass`, and most of the rest of the code will stay the same.